docs/developer-guide/contributing.md

```diff
@@ -239,6 +239,34 @@ If you touched UI code, you should also run the Yarn linter on it:
 * Run `make lint-ui`
 * Fix any of the errors reported by it
 
+## Contributing to Argo CD UI
+
+Argo CD, along with Argo Workflows, uses shared React components from [Argo UI](https://github.com/argoproj/argo-ui). Examples of some of these components include buttons, containers, form controls, 
+and others. Although you can make changes to these files and run them locally, in order to have these changes added to the Argo CD repo, you will need to follow these steps. 
+
+1. Fork and clone the [Argo UI repository](https://github.com/argoproj/argo-ui).
+
+2. `cd` into your `argo-ui` directory, and then run `yarn install`. 
+
+3. Make your file changes.
+
+4. Run `yarn start` to start a [storybook](https://storybook.js.org/) dev server and view the components in your browser. Make sure all your changes work as expected. 
+
+5. Use [yarn link](https://classic.yarnpkg.com/en/docs/cli/link/) to link Argo UI package to your Argo CD repository. (Commands below assume that `argo-ui` and `argo-cd` are both located within the same parent folder)
+
+    * `cd argo-ui`
+    * `yarn link`
+    * `cd ../argo-cd`
+    * `yarn link argo-ui`
+
+    Once `argo-ui` package has been successfully linked, test out changes in your local development environment. 
+
+6. Commit changes and open a PR to [Argo UI](https://github.com/argoproj/argo-ui). 
+
+7. Once your PR has been merged in Argo UI, change the commit SHA in the `ui/yarn.lock` file in Argo CD. (i.e. `resolved "https://github.com/argoproj/argo-ui.git#[commit SHA]"`) 
```

Comment:
Instead of manually editing `ui/yarn.lock` you can run ` yarn add https://github.com/argoproj/argo-ui.git` in `ui` directory. It will update dependency to the latest master commit.

Reply:
oh cool I didn't know that! Just updated it :)